dandiapi/api/management/commands/correct_metadata.py

```diff
@@ -0,0 +1,134 @@
+from __future__ import annotations
+
+from copy import deepcopy
+from typing import TYPE_CHECKING, Any
+
+if TYPE_CHECKING:
+    from collections.abc import Callable
+
+from django.db import transaction
+import djclick as click
+
+from dandiapi.api.models import Version
+from dandiapi.api.tasks import write_manifest_files
+
+
+@click.command(
+    help='Correct corrupted metadata. If `--dandiset` and `--dandiset-version` are provided, '
+    'apply the correction to the specified Dandiset version. Otherwise, apply the correction to '
+    'all Dandiset versions.'
+)
+@click.option(
+    '--dandiset',
+    'dandiset',
+    help='The Dandiset. (This must be provided if `--dandiset-version` is provided.)',
+)
+@click.option(
+    '--dandiset-version',
+    'dandiset_version',
+    help='The version of the Dandiset. (This must be provided if `--dandiset` is provided.)',
+)
+def correct_metadata(*, dandiset: str | None, dandiset_version: str | None):
+    # Func to use to correct the metadata (update as needed)
+    correct_func: Callable[[dict], dict | None] = correct_affiliation_corruption
+
+    # Ensure both options are provided together (or both omitted)
+    if (dandiset is None) != (dandiset_version is None):
+        raise click.UsageError(
+            'Both `--dandiset` and `--dandiset-version` must be provided together.'
+        )
+
+    vers = (
+        Version.objects.all()
+        if dandiset is None
+        else (Version.objects.get(dandiset=dandiset, version=dandiset_version),)
+    )
+
+    # List of changes to be applied to the database represented as tuples of the form
+    # version to be changed and the new metadata
+    changes: list[tuple[Version, dict]] = []
+    for ver in vers:
+        try:
+            meta_corrected = correct_func(ver.metadata)
+        except Exception as e:
+            click.echo(
+                f'{ver.dandiset.identifier}/{ver.version}: Failed to correct metadata with '
+                f'`{correct_func.__name__}()`'
+            )
```

Comment:
Have you actually observed any exceptions raised when performing the metadata correction?

Reply:
No, I have not. The exception handling is put in as a protective programming measure, particularly to handle situation when `correct_func` is assigned to some other function, possibly in the future.